### PR TITLE
feat: add verifiable build to ci

### PR DESCRIPTION
**Pull request by keyvankhademi**

### CODE REVIEW

File: .github/workflows/anchor.yml [hunk 1]
```diff
@@ -25,9 +25,17 @@ jobs:
       - name: Install Anchor
         working-directory: ./staking
         run: npm i -g @coral-xyz/anchor-cli@0.27.0
+      - name: Build anchor (verifiable)
+        working-directory: ./staking
+        run: |
+          npm run build_wasm
```
Comment (guibescos) [inline .github/workflows/anchor.yml:31]: Any reason why these commands are not in the script `test:ci` ?

File: staking/scripts/build_verifiable_staking_program.sh [hunk 2]
```diff
@@ -5,14 +5,37 @@ set -euo pipefail
...
+    echo "Building the image for the staking program test"
+    docker build --platform linux/amd64 --build-arg TEST=true -t staking-build -f "$REPO_ROOT"/staking/Dockerfile "$REPO_ROOT"/staking
+else
+    echo "Building the image for the staking program production"
+    docker build --platform linux/amd64 -t staking-build -f "$REPO_ROOT"/staking/Dockerfile "$REPO_ROOT"/staking
+fi
 
 echo "Building the staking program"
-docker run --platform linux/x86_64 --rm -v "$REPO_ROOT"/staking/artifacts:/artifacts staking-build
+docker run --platform linux/amd64 --rm -v "$REPO_ROOT"/staking/artifacts:/artifacts staking-build
```
Comment (guibescos) [inline staking/scripts/build_verifiable_staking_program.sh:35]: Why this change?

File: .github/workflows/anchor.yml [hunk 3]
```diff
@@ -25,9 +25,17 @@ jobs:
...
+          ./scripts/build_verifiable_staking_program.sh -t
+          cp -r ./artifacts/target/* ./target/
+          ls -al target
```
Comment (guibescos) [inline .github/workflows/anchor.yml:34]: This `ls` seems doesn't seem useful

File: staking/Dockerfile [hunk 4]
```diff
@@ -6,9 +6,11 @@
...
 
+ARG TEST=false
+ENV TEST_ENV=$TEST
 WORKDIR /workspace
 
 COPY . .
 
 CMD ["bash", "-c", \
-        "anchor build && cp target/deploy/staking.so /artifacts/staking.so"]
+        "if [ \"$TEST_ENV\" = \"true\" ]; then anchor build -- --features mock-clock; else anchor build; fi && if [ \"$TEST_ENV\" = \"true\" ]; then chmod -R 777 target/; fi && cp -r target/ /artifacts"]
```
Comment (guibescos) [inline staking/Dockerfile:16]: is the `chmod` needed here?

File: staking/scripts/build_verifiable_staking_program.sh (hunk 2 above)
Comment (keyvankhademi) [reply staking/scripts/build_verifiable_staking_program.sh]: I believe `x86_64` is just an alias for `amd64`. I see that everywhere is using `amd64` now so I thought we go with that.

File: .github/workflows/anchor.yml (hunk 3 above)
Comment (keyvankhademi) [reply .github/workflows/anchor.yml]: right😅

File: staking/Dockerfile (hunk 4 above)
Comment (keyvankhademi) [reply staking/Dockerfile]: Yes we run into permission errors without it. because the files are created by the docker user.

File: .github/workflows/anchor.yml [hunk 5]
```diff
@@ -25,9 +25,17 @@ jobs:
       - name: Install Anchor
         working-directory: ./staking
         run: npm i -g @coral-xyz/anchor-cli@0.27.0
+      - name: Build anchor (verifiable)
+        working-directory: ./staking
+        run: |
+          npm run build_wasm
+          ./scripts/build_verifiable_staking_program.sh -t
+          cp -r ./artifacts/target/* ./target/
```
[the remark this answers is not in the document]
Comment (keyvankhademi) [reply .github/workflows/anchor.yml]: I was also thinking about just voluming the target folder. Not sure if there are any downsides for that. I gave it a try.

File: .github/workflows/anchor.yml (hunk 1 above)
Comment (keyvankhademi) [reply .github/workflows/anchor.yml]: yeah it makes sense to do that. Fixed.

**Review by guibescos — APPROVED** [review]
